client-app/pages/external/slug-content.vue

```diff
@@ -0,0 +1,86 @@
+<template>
+  <div v-if="!loading && (hasContent || objectType)" class="slug-content">
+    <component :is="CategoryComponent" v-if="objectType === 'Category'" :category-id="slugInfo?.entityInfo?.objectId" />
```

Comment:
What is an advantage of using dynamic components here? `CategoryComponent`, `Product` and `StaticPage` are always the same components.

Reply:
it is for chunks https://vuejs.org/guide/components/async

Reply:
yes, that's fine about async components, perfect solution. I meant using `<component :is="...`

it aims to dynamically render a component depends on the variable passed in `is`. But here we can use `<CategoryComponent ...` right away, since after loading it will never be changed.